src/Engine/Physics/Shape.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <glm/vec3.hpp>
+
+class btCollisionShape;
+
+namespace Physics {
+
+    /// Represents a shape for physics objects and facilitates creation of
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') class 'Shape' does not have a copy constructor which is recommended since the class contains a pointer to allocated memory. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AnoCopyConstructor)